File: leetcode/2391-minimum-amount-of-time-to-collect-garbage/garbageCollection.test.ts
import cases from "./cases.json";
import garbageCollection from "./garbageCollection";

test("minimum amount of time to collect garbage", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = garbageCollection(
      testCase.input.garbage,
      testCase.input.travel
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});

File: leetcode/2391-minimum-amount-of-time-to-collect-garbage/garbageCollection_test.rs
use std::{env::current_dir, path::Path};

use serde::Deserialize;

use crate::{util::load_cases, Solution};

#[derive(Deserialize, Debug)]
struct Input {
    garbage: Vec<String>,
    travel: Vec<i32>,
}

#[derive(Deserialize, Debug)]
struct Case {
    input: Input,
    output: i32,
}

#[test]
fn test_minimum_amount_of_time_to_collect_garbage() {
    let current_path = current_dir().unwrap().join(file!());
    let dir = current_path.parent().unwrap();

    let cases: Vec<Case> = load_cases(Path::new(&dir.join("./cases.json")));

    for case in &cases {
        assert_eq!(
            Solution::garbage_collection(case.input.garbage.clone(), case.input.travel.clone()),
            case.output
        );
    }
}
